### PR TITLE
Vidtodo [Add new Extractor]

### PR DESCRIPTION
**Pull request by alrii**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [x] New extractor
- [x] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

File: youtube_dl/extractor/vidtodo.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            raise ExtractorError('File not found', expected=True, video_id=video_id)
+
+        source = re.search(r'\.setup\((.+?\],image:.+?\")', codes).group(1) + '}'
+        source_keys = ['sources', 'file', 'label', 'image']
+        for key_ in source_keys:
+            source = source.replace(key_, '"' + key_ + '"')
+        source = eval(source)
+
+        formats = []
+        for format_ in source['sources']:
```
Comment (dstftw) [inline youtube_dl/extractor/vidtodo.py]: `_parse_jwplayer_data`.

File: youtube_dl/extractor/vidtodo.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            'id': '4c8rx0tt8ek4',
+            'ext': 'mp4',
+            'title': 'Watch 343291981 mp4',
+        },
+    },]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        std_headers['referer'] = 'https://rg3.github.io/youtube-dl/'
```
[the remark this answers is not in the document]
Comment (alrii) [reply youtube_dl/extractor/vidtodo.py]: if I remove referer test doesn't work. Can I use different url instead?

[the remark this answers is not in the document]
Comment (dstftw) [reply youtube_dl/extractor/vidtodo.py]: I'm not about the URL. **Do not touch** the global `std_headers`.

**Comment by alrii** [comment]
I made revision according to your review. Can anyone assess my changes?

**Comment by dstftw** [comment]
No longer available.

**Pull request closed without merging**